gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java

```diff
@@ -244,97 +229,122 @@ public Single<Certificate> create(String domain, NewCertificate newCertificate,
+                        .map(schema -> new CertificateWithSchema(certificate, objectMapper.readValue(schema, CertificateSchema.class))))
                 .flatMap(oldCertificate -> {
-                    boolean oldWithMTls = usageContains(oldCertificate.getCertificate().getConfiguration(), "mtls");
+                    boolean oldWithMTls = usageContains(oldCertificate.certificate().getConfiguration(), "mtls");
                     boolean newWithMTls = usageContains(updateCertificate.getConfiguration(), "mtls");
                     if (oldWithMTls && !newWithMTls){
-                        return checkIdentityProviderUsage(oldCertificate.getCertificate())
+                        return checkIdentityProviderUsage(oldCertificate.certificate())
                                 .map(cert -> oldCertificate);
                     } else {
                         return Single.just(oldCertificate);
                     }
                 })
                 .flatMap(oldCertificate -> {
-                    Certificate certificateToUpdate = new Certificate(oldCertificate.getCertificate());
-                    certificateToUpdate.setName(updateCertificate.getName());
-                    certificateToUpdate.setUpdatedAt(new Date());
-                    if (!certificateToUpdate.isSystem()) { // system certificate can't be updated
-                        try {
-                            CertificateSchema certificateSchema = oldCertificate.getSchema();
-                            JsonNode oldCertificateConfiguration = objectMapper.readTree(oldCertificate.getCertificate().getConfiguration());
-                            JsonNode certificateConfiguration = objectMapper.readTree(updateCertificate.getConfiguration());
-                            var key = certificateSchema.getProperties()
-                                    .entrySet()
-                                    .stream()
-                                    .filter(map -> map.getValue().getWidget() != null && "file".equals(map.getValue().getWidget()))
-                                    .map(Map.Entry::getKey)
-                                    .findFirst().orElse(null);
-                            String oldFileInformation = oldCertificateConfiguration.get(key).asText();
-                            String fileInformation = certificateConfiguration.get(key).asText();
-                            // file has changed, let's update it
-                            if (!oldFileInformation.equals(fileInformation)) {
-                                JsonNode file = objectMapper.readTree(certificateConfiguration.get(key).asText());
-                                byte[] data = Base64.getDecoder().decode(file.get(CONTENT).asText());
-                                certificateToUpdate.setMetadata(Collections.singletonMap(CertificateMetadata.FILE, data));
```

Comment:
I am not sure how it has worked with immutable map

In AbstractCertificateProvider is executed
  // update metadata
certificateMetadata.getMetadata().put(CertificateMetadata.DIGEST_ALGORITHM_NAME, signature.getDigestName());

Reply:
Indeed, this is strange but I'm ok with your change to move it to non singleton HashMap